### PR TITLE
Use top level program for the Worker template

### PR DESCRIPTION
**Pull request by IEvangelist**
Reduce the Worker template by half, using top level statements.

/cc @bradygaster and @shirhatti

### CODE REVIEW

**Comment by bradygaster** [comment]
Timing seemed appropriate with other discussions going on. I'd love to see the Worker as a top-level. I like this.

**Comment by shirhatti** [comment]
I'd like to see the template further simplified with [global usings  (like in the websdk)](https://github.com/dotnet/sdk/pull/18459).

@JunTaoLuo  👀

File: src/ProjectTemplates/Web.ProjectTemplates/content/Worker-CSharp/Program.cs [hunk 1]
```diff
@@ -1,24 +1,12 @@
...
-using System.Linq;
-using System.Threading.Tasks;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
+using Company.Application1;
 
-namespace Company.Application1
-{
-    public class Program
+using IHost host = Host.CreateDefaultBuilder(args)
```
Comment (davidfowl) [inline src/ProjectTemplates/Web.ProjectTemplates/content/Worker-CSharp/Program.cs:5]:
```suggestion
IHost host = Host.CreateDefaultBuilder(args)
```

I'd remove the using here. RunAsync will dispose (asynchronously).

**Comment by davidfowl** [comment]
I think we want to do this for all of the templates as well.

File: src/ProjectTemplates/Web.ProjectTemplates/content/Worker-CSharp/Program.cs (hunk 1 above)
Comment (IEvangelist) [reply src/ProjectTemplates/Web.ProjectTemplates/content/Worker-CSharp/Program.cs]: I accepted the suggestion, even though I'm not a fan of it. It reminds me of the `public static IHostBuilder CreateHostBuilder(string[] args)` in that it's "magical", and does more than it _should_. I assume that other developers would see this and ask, "why are we not wrapping this `IDisposable` instance in a `using`?". In other words, it's not obvious that `host.RunAsync()` manages the disposable of itself.

**Comment by shirhatti** [comment]
FYI @DamianEdwards since he's updating templates right now. I'm okay merging this as is and letting Damian make any additional changes in his PR

**Comment by DamianEdwards** [comment]
#33947

**Review by bradygaster — APPROVED** [review]

**Pull request merged**